pkg/resources/nodeportproxy/nodeportproxy.go

```diff
@@ -418,13 +419,19 @@ func FrontLoadBalancerServiceCreator(data *resources.TemplateData) reconciling.N
 
 			if data.Seed().Spec.NodeportProxy.Envoy.LoadBalancerService.SourceRanges != nil {
 				for _, cidr := range data.Seed().Spec.NodeportProxy.Envoy.LoadBalancerService.SourceRanges {
-					s.Spec.LoadBalancerSourceRanges = append(s.Spec.LoadBalancerSourceRanges, string(cidr))
+					if !slices.Contains(s.Spec.LoadBalancerSourceRanges, string(cidr)) {
+						s.Spec.LoadBalancerSourceRanges = append(s.Spec.LoadBalancerSourceRanges, string(cidr))
+					}
 				}
 			}
 
 			// Check if allowed IP ranges are configured and set the LoadBalancer source ranges
 			if data.Cluster().Spec.APIServerAllowedIPRanges != nil {
-				s.Spec.LoadBalancerSourceRanges = append(s.Spec.LoadBalancerSourceRanges, data.Cluster().Spec.APIServerAllowedIPRanges.CIDRBlocks...)
+				for _, cidr := range data.Cluster().Spec.APIServerAllowedIPRanges.CIDRBlocks {
+					if !slices.Contains(s.Spec.LoadBalancerSourceRanges, cidr) {
+						s.Spec.LoadBalancerSourceRanges = append(s.Spec.LoadBalancerSourceRanges, cidr)
+					}
+				}
```

Comment:
This can be solved by using apimaachinery's sets.String type to just create a string set from the cluster spec and then assining .List() to the service's spec. And for the other one ([]CIDRBlock) it might also be nice to use the string sets, just so we don't have one Go dependency just for slice.Contains.